docs/USAGE-GUIDE.md

```diff
@@ -464,103 +464,25 @@ Client authentication can be configured by calling `s2n_config_set_client_auth_t
 
-### s2n\_config\_set\_protocol\_preferences
+### OCSP Stapling
 
-```c
-int s2n_config_set_protocol_preferences(struct s2n_config *config,
-                                        const char **protocols,
-                                        int protocol_count);
-```
-
-**s2n_config_set_protocol_preferences** sets the application protocol
-preferences on an **s2n_config** object.  **protocols** is a list in order of
-preference, with most preferred protocol first, and of length
-**protocol_count**.  When acting as an **S2N_CLIENT** the protocol list is
-included in the Client Hello message as the ALPN extension.  As an
-**S2N_SERVER**, the list is used to negotiate a mutual application protocol
-with the client. After the negotiation for the connection has completed, the
-agreed upon protocol can be retrieved with [s2n_get_application_protocol](#s2n_get_application_protocol)
+Online Certificate Status Protocol (OCSP) is a protocol to establish whether or not a certificate has been revoked. The requester (usually a client), asks the responder (usually a server), to ‘staple’ the certificate status information along with the certificate itself. The certificate status sent back will be either expired, current, or unknown, which the requester can use to determine whether or not to accept the certificate.
 
-### s2n\_config\_set\_extension\_data
+OCSP stapling can be applied to both client and server certificates when using TLS1.3, but only to server certificates when using TLS1.2.
 
-```c
-int s2n_config_set_extension_data(struct s2n_config *config, s2n_tls_extension_type type, const uint8_t *data, uint32_t length);
-```
+To use OCSP stapling, both server and client must call `s2n_config_set_status_request_type()` with S2N_STATUS_REQUEST_OCSP. The server (or client, if using client authentication) will also need to call `s2n_config_set_extension_data()` with S2N_EXTENSION_OCSP_STAPLING to set the raw bytes of the OCSP stapling data. Note that `s2n_config_set_extension_data()` modifies the certificate chain instead of the config, so it can’t be used with configs that share a certificate chain with other configs. This means that when using OCSP data, `s2n_config_add_cert_chain_and_key()` must be used to set the certificate chain, NOT `s2n_config_add_cert_chain_and_key_to_store()`.
 
-**s2n_config_set_extension_data** Sets the extension data in the **s2n_config**
-object for the specified extension.  This method will clear any existing data
-that is set.   If the data and length parameters are set to NULL, no new data
-is set in the **s2n_config** object, effectively clearing existing data.
+The OCSP stapling information can be automatically validated by the underlying libcrypto if `s2n_config_set_check_stapled_ocsp_response()` is called with “1”. Call `s2n_connection_get_ocsp_response()` to retrieve the received OCSP stapling information for manual verification.
```

Comment:
I just double checked this. We [check the OCSP response by default](https://github.com/aws/s2n-tls/blob/cd9190c4c6648e9477d4efc3d3032000a8477cc3/tls/s2n_config.c#L103), and s2n_config_set_check_stapled_ocsp_response would likely only be used to disable that check.